python/remage/geombench/cli.py

```diff
@@ -0,0 +1,214 @@
+)
+from .summary_generator import SummaryGenerator
+
+
+def generate_macro(args, output_file_stem: str = "") -> str:
+    """Generate a macro file for the remage geometry benchmark.
+
+    This function creates a macro file based on the provided command-line
+    arguments, which will be used to configure the remage simulation for the
+    geometry benchmark.
+
+    Parameters
+    ----------
+    args
+        Parsed command-line arguments.
+        Consisting of:
+        - geometry: Path to the geometry file.
+        - output_dir: Directory to store benchmark output files.
+        - num_events: Number of events to simulate.
+        - grid_increment: Increment between grid points in mm.
+        - grid_increments: Optional dict of increments per dimension in mm.
+        - dry_run: If set, only generate the macro file without running remage.
+
+    Returns
+    -------
+    str
+        Path to the generated macro file.
+    """
+
+    geometry_path = Path(args.geometry)
+    filename = geometry_path.absolute()
+    output_dir = Path(args.output_dir)
+    output_file = output_dir / (output_file_stem + ".lh5")
+    if issubclass(type(args.grid_increments), str) and args.grid_increments:
+        grid_increments = ast.literal_eval(args.grid_increments)
+        increment_x = grid_increments.get("x", args.grid_increment)
+        increment_y = grid_increments.get("y", args.grid_increment)
+        increment_z = grid_increments.get("z", args.grid_increment)
+    else:
+        increment_x = args.grid_increment
+        increment_y = args.grid_increment
+        increment_z = args.grid_increment
+    n_events = args.num_events
+
+    commands = [
+        f"/RMG/Geometry/IncludeGDMLFile {filename}",
+        "/RMG/Output/ActivateOutputScheme GeomBench",
+        f"/RMG/Output/FileName {output_file}",
+        "/run/initialize",
+        "/RMG/Generator/Select GeomBench",
+        f"/RMG/Generator/Benchmark/IncrementX {increment_x} mm",
+        f"/RMG/Generator/Benchmark/IncrementY {increment_y} mm",
+        f"/RMG/Generator/Benchmark/IncrementZ {increment_z} mm",
+        f"/run/beamOn {n_events}",
+    ]
+    macro_content = "\n".join(commands)
+
+    if args.dry_run:
+        logger = rmg_logging.setup_log()
+        logger.info("Dry run enabled. Generated macro content:")
+        logger.info(macro_content)
+        return ""
+
+    return macro_content
+
+
+def remage_geombench_cli(external_args: list[str] | None = None) -> int:
+    """Command-line interface for the remage geometry benchmark.
+
+    This function parses command-line arguments, sets up the macros, runs
+    the geometry benchmark using the remage simulation framework, and analyzes
+    the output plus provides a summary of the results.
+
+    Returns
+    -------
+    int
+        Exit code of the program. Returns 0 on success, non-zero on failure.
+    """
+    parser = argparse.ArgumentParser(
+        description="Run remage geometry benchmark and analyze results."
```

Comment:
```suggestion
        description="Run remage geometry benchmark and analyze results.",
        allow_abbrev=False,
```

to match the behaviour of the main CLI